src/util.js

```diff
@@ -132,15 +132,19 @@ export default class Utils {
         res => {
           // explicitly treat incoming data as utf8 (avoids issues with multi-byte chars)
           res.setEncoding('utf8');
-
+          let statusCode = res.statusCode;
           // incrementally capture the incoming response body
           let body = '';
           res.on('data', chunk => {
             body += chunk;
           });
 
           res.on('end', () => {
-            success(body);
+            if (statusCode == 200) {
```

Comment:
Why use a local var instead of `res.statusCode`?

Reply:
You may also want to check [`res.complete`](https://nodejs.org/api/http.html#http_message_complete).

Reply:
Also, because the `error` callback is called from the [stream `'error'` event](https://nodejs.org/api/stream.html#stream_event_error_1), it seems like it's usually called with an `Error` object.

So, it might make sense to generate pass an error into the callback rather than a string.

```js
const err = new Error(`HTTP GET failed: ${res.statusMessage}`);
err.code = res.statusMessage;
err.response = res;
error(err);
```

Or maybe something simpler:

```js
error(new Error(`HTTP GET failed: ${res.statusCode}`));
```

Reply:
Updated